smart_open/tests/test_smart_open.py

```diff
@@ -886,21 +947,18 @@ class CompressionFormatTest(unittest.TestCase):
     Test that compression
     """
 
-    TEXT = 'Hello'
-
     def write_read_assertion(self, suffix):
         with tempfile.NamedTemporaryFile('wb', suffix=suffix, delete=False) as infile:
             test_file = infile.name
 
-        text = self.TEXT.encode('utf8')
         with smart_open.smart_open(test_file, 'wb') as fout:  # 'b' for binary, needed on Windows
-            fout.write(text)
+            fout.write(SAMPLE_BYTES)
```

Comment:
Why is it necessary for this test to write to a temporary file? Is using an in-memory buffer not sufficient for this test?

I'm not keen on unit tests that have side effects like writing to local storage. Sometimes its a necessary evil, but it smells more like an integration test than a unit test.